src/app/snippet/services/snippet-context.service.ts

```diff
@@ -0,0 +1,20 @@
+import {Injectable} from '@angular/core';
+import {Observable, Subject} from "rxjs";
+import {Snippet} from "../models/snippet";
+
+@Injectable() // Service injecté au niveau du module "Snippet"
+export class SnippetContextService {
+
+  private snippetSubject = new Subject<Snippet>();
+
+  constructor() { }
+
+  set sendSnippetSubject(snippet: Snippet) {
+    this.snippetSubject.next(snippet);
+  }
+
+  get snippetSubjectToObservable$(): Observable<Snippet> {
```

Comment:
Qu'est-ce que tu pense de :
```suggestion
  get selectedSnippet$(): Observable<Snippet> {
```

Reply:
Oui, bien ça. :)